component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/jwt/JWTBearerGrantHandler.java

```diff
@@ -758,12 +882,101 @@ protected X509Certificate resolveSignerCertificate(JWSHeader header,
      * @param customClaims a map of custom claims
      * @return whether the token is valid based on other claim values
      */
-    protected boolean validateCustomClaims(Map< String, Object > customClaims) {
+    protected boolean validateCustomClaims(Map<String, Object> customClaims) {
+
         return true;
     }
 
     private void handleException(String errorMessage) throws IdentityOAuth2Exception {
+
         log.error(errorMessage);
         throw new IdentityOAuth2Exception(errorMessage);
     }
-}
\ No newline at end of file
+
+    private void logAndThrowException(String errorMessage, Exception e) throws IdentityOAuth2Exception {
+
+        log.error(errorMessage);
```

Comment:
do we need to log it here? Since we are also throwing the exception, caller should log it. Otherwise we'll get the same stacktrace duplicated.